src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -104,9 +105,20 @@
     private static final Set<Integer> TOKEN_TYPES_WITH_TEXT_ATTRIBUTE =
         Stream.of(
             TokenTypes.IDENT, TokenTypes.STRING_LITERAL, TokenTypes.CHAR_LITERAL,
-            TokenTypes.NUM_LONG, TokenTypes.NUM_INT, TokenTypes.NUM_DOUBLE, TokenTypes.NUM_FLOAT)
+            TokenTypes.NUM_LONG, TokenTypes.NUM_INT, TokenTypes.NUM_DOUBLE, TokenTypes.NUM_FLOAT,
+            TokenTypes.TEXT_BLOCK_CONTENT)
         .collect(Collectors.toSet());
 
+    /**
+     * This regexp is used to convert new line to newline tag.
+     */
+    private static final Pattern NEWLINE_TO_TAG = Pattern.compile("[\n]");
```

Comment:
why not just `Pattern.compile("\n")` ?

Reply:
if we do simple substitution of string to string can we use `replace(CharSequence target, CharSequence replacement)` ?

Reply:
Sir, simple string substitutions are prohibited by teamCity build. TeamCity suggests that `String.replaceAll()` should be replace with `Pattern.compile()` so this change.

Reply:
from IDEA inspection
```
Java
Performance
Dynamic regular expression could be replaced by compiled Pattern
XpathUtil
'replace()' could be replaced with compiled 'java.util.regex.Pattern' construct
```

but code java do nothing with dynamic regexp in java9 and above
http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/d966fc5a7be5/src/java.base/share/classes/java/lang/String.java#l2178

but here I see it - jdk8:
http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/String.java#l2225

Reply:
lets keep regexp

Reply:
Done, I was doing some test since I was bit skeptical that would it change something in appropriately within a string, but it`s working fine as anyway it would get converted to `////n` when present within a String.